test/suites/termination/termination_test.go

```diff
@@ -47,4 +51,112 @@ var _ = Describe("Termination", func() {
 			g.Expect(lo.FromPtr(env.GetInstanceByID(instanceID).State.Name)).To(BeElementOf("terminated", "shutting-down"))
 		}, time.Second*10).Should(Succeed())
 	})
+	// Pods from Karpenter nodes are expected to drain in the following order:
+	//   1. Non-Critical Non-Daemonset pods
+	//   2. Non-Critical Daemonset pods
+	//   3. Critical Non-Daemonset pods
+	//   4. Critical Daemonset pods
+	// Pods in one group are expected to be fully removed before the next group is executed
+	It("should drain pods on a node in order", func() {
```

Comment:
Should we also include `system-cluster-critical` pods and daemonsets like in the functional tests upstream?

Reply:
Good callout -- will add